### ISSUE TITLE
 Incorrect Results When Running Pre-trained Model Locally

### ISSUE THREAD

**Issue by C-hongfei** [lpiccinelli-eth/UniDepth#80]
Hi, thank you for your excellent work on this project! 
When I downloaded the pre-trained model to my local environment and ran the demo, I encountered incorrect results. The ARel metric is showing 72.07%, which seems to be much higher than expected. Below are the changes I made. Could you please help me identify what might be causing this issue?
Changes Made:
hubconf.py,line33,
`# path = huggingface_hub.hf_hub_download(repo_id=f"lpiccinelli/unidepth-{version}-{backbone}", filename=f"pytorch_model.bin", repo_type="model")
        path = f"lpiccinelli/unidepth-{version}-{backbone}"+"/pytorch_model.bin"`
unidepth/models/backbones/dinov2.py, line451,
`    elif pretrained is not None:
        state_dict = torch.load('dinov2_vits14_pretrain.pth', map_location="cpu")
        info = model.load_state_dict(state_dict, strict=False)`

![output](https://github.com/user-attachments/assets/72c9adfd-6496-4814-b384-4c84547aca2b)

**Comment by lpiccinelli-eth** [comment]
That output looks like the weights are initialized from random instead of being loaded from the checkpoint.
Can you try to see what is contained in `info`? That returns the info about the checkpoint loading and should return "all keys matched successfully"

**Comment by C-hongfei** [comment]
> That output looks like the weights are initialized from random instead of being loaded from the checkpoint. Can you try to see what is contained in `info`? That returns the info about the checkpoint loading and should return "all keys matched successfully"

Thank you.
You're right, it shows that many keys are missing, which might be due to issues with loading the offline model. But after using the update_v2 branch, everything works normally.
